share/functions/help.fish

```diff
@@ -193,7 +194,7 @@ function help --description 'Show help for the fish shell'
     else
         # Work around lynx bug where <div class="contents"> always has the same formatting as links (unreadable)
         # by using a custom style sheet. See https://github.com/fish-shell/fish-shell/issues/4170
-        if string match -q 'lynx*' -- $fish_browser
+        if string match -q '^lynx' -- $fish_browser
```

Comment:
If you use `^`, you need to use the `--regex`/`-r` flag, by default `string match` will do glob-style matching.

Reply:
My bad, didn't see it was `-q` not `-qr` so I read it as a regex, should I revert this change or switch to `-qr` ?

Reply:
Both work fine, you choose :)

Reply:
Well, since all but one `string match` uses regex matching I'll switch to `-qr` to homogenize.